phase4-profile-entsog/src/main/java/com/helger/phase4/profile/entsog/ENTSOGCompatibilityValidator.java

```diff
@@ -0,0 +1,306 @@
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.helger.phase4.profile.entsog;
+
+import javax.annotation.Nonnull;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.helger.commons.ValueEnforcer;
+import com.helger.commons.error.IError;
+import com.helger.commons.error.SingleError;
+import com.helger.commons.error.list.ErrorList;
+import com.helger.commons.string.StringHelper;
+import com.helger.phase4.attachment.EAS4CompressionMode;
+import com.helger.phase4.crypto.ECryptoAlgorithmCrypt;
+import com.helger.phase4.crypto.ECryptoAlgorithmSign;
+import com.helger.phase4.crypto.ECryptoAlgorithmSignDigest;
+import com.helger.phase4.ebms3header.Ebms3AgreementRef;
+import com.helger.phase4.ebms3header.Ebms3From;
+import com.helger.phase4.ebms3header.Ebms3SignalMessage;
+import com.helger.phase4.ebms3header.Ebms3To;
+import com.helger.phase4.ebms3header.Ebms3UserMessage;
+import com.helger.phase4.mgr.MetaAS4Manager;
+import com.helger.phase4.model.EMEP;
+import com.helger.phase4.model.EMEPBinding;
+import com.helger.phase4.model.pmode.IPMode;
+import com.helger.phase4.model.pmode.PModePayloadService;
+import com.helger.phase4.model.pmode.PModeValidationException;
+import com.helger.phase4.model.pmode.leg.EPModeSendReceiptReplyPattern;
+import com.helger.phase4.model.pmode.leg.PModeLeg;
+import com.helger.phase4.model.pmode.leg.PModeLegErrorHandling;
+import com.helger.phase4.model.pmode.leg.PModeLegProtocol;
+import com.helger.phase4.model.pmode.leg.PModeLegSecurity;
+import com.helger.phase4.profile.IAS4ProfileValidator;
+import com.helger.phase4.soap.ESoapVersion;
+import com.helger.phase4.wss.EWSSVersion;
+
+/**
+ * Validate certain requirements imposed by the ENTSOG project.
+ *
+ * @author Pavel Rotek
+ */
+public class ENTSOGCompatibilityValidator implements IAS4ProfileValidator {
+  @SuppressWarnings("unused")
+  private static final Logger LOGGER = LoggerFactory.getLogger(ENTSOGCompatibilityValidator.class);
+
+  public ENTSOGCompatibilityValidator() {
+  }
+
+  @Nonnull
+  private static IError _createError(@Nonnull final String sMsg) {
+    return SingleError.builderError().setErrorText(sMsg).build();
+  }
+
+  private void _checkIfLegIsValid(@Nonnull final IPMode aPMode,
+                                  @Nonnull final ErrorList aErrorList,
+                                  @Nonnull final PModeLeg aPModeLeg) {
+    final PModeLegProtocol aLegProtocol = aPModeLeg.getProtocol();
+    if (aLegProtocol == null) {
+      aErrorList.add(_createError("PMode Leg 1 is missing Protocol"));
```

Comment:
I just recently made some improvements to the error messages. I would apply this afterwards myself if you don't mind.

Reply:
yes, sure